src/components/Calulator.js

```diff
@@ -2,78 +2,126 @@
 
 import { Component } from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
   }
 
-  render() {
-    return (
-      <div className="calculator">
-        <div className="box">
-          <div className="result four-cell">0</div>
-          <button type="button" className="key">
-            AC
-          </button>
-          <button type="button" className="key">
-            +/-
-          </button>
-          <button type="button" className="key">
-            %
-          </button>
-          <button type="button" className="key operator">
-            /
-          </button>
-          <button type="button" className="key">
-            7
-          </button>
-          <button type="button" className="key">
-            8
-          </button>
-          <button type="button" className="key">
-            9
-          </button>
-          <button type="button" className="key operator">
-            *
-          </button>
-          <button type="button" className="key">
-            4
-          </button>
-          <button type="button" className="key">
-            5
-          </button>
-          <button type="button" className="key">
-            6
-          </button>
-          <button type="button" className="key operator">
-            -
-          </button>
-          <button type="button" className="key">
-            1
-          </button>
-          <button type="button" className="key">
-            2
-          </button>
-          <button type="button" className="key">
-            3
-          </button>
-          <button type="button" className="key operator">
-            +
-          </button>
-          <button type="button" className="key two-cell">
-            0
-          </button>
-          <button type="button" className="key">
-            .
-          </button>
-          <button type="button" className="key operator">
-            =
-          </button>
-        </div>
-      </div>
-    );
-  }
+ handleResult = (e) => {
+   const name = e.target.textContent;
+   this.setState((prev) => ({
+     ...prev,
+     total: name,
+   }));
+ };
+
+ handleKeyPress = (e) => {
+   const keyValue = e.target.textContent;
+   const result = calculate(this.state, keyValue);
+   this.setState(result);
+ };
+
+ render() {
+   const { total, next, operation } = this.state;
+   return (
+     <div className="calculator">
+       <div className="box">
+         <div className="result four-cell" onChange={this.handleClick}>
+           {total ?? 0}
+           {operation}
+           {next}
+         </div>
```

Comment:
- [ ] Great job adding this functionality, but right now the logic is a little buggy. Kindly fix the issues mentioned below
  - Please ensure that when the user adds a negative value from the start, it does not break the calculated. Let's say I select -3 at the start, it will stop the calculator from functioning
![image](https://user-images.githubusercontent.com/90556221/188847008-adacebed-ae48-44f4-97f3-d7715e99e779.png)

  - Please ensure that the multiplication & division logic work as expected. Right now none of those two operations work as expected
![image](https://user-images.githubusercontent.com/90556221/188847520-11996a44-a593-48cd-81ca-95e5024388b8.png)

Reply:
Thank you for your feedback, I will modify It.